internal/command/command.go

```diff
@@ -355,10 +356,246 @@ var CmdUpdateApis = &Command{
+		repoPath = filepath.Join(tmpRoot, fmt.Sprintf("google-cloud-%s", flagLanguage))
+	}
+
+	return languageRepo, repoPath, inputDir, nil
+}
+
+func checkFlags() error {
+	if !supportedLanguages[flagLanguage] {
+		return fmt.Errorf("invalid -language flag specified: %q", flagLanguage)
+	}
+	if flagImage == "" {
+		return fmt.Errorf("Create release pr requires image")
+	}
+	return nil
+}
+
+func generateReleasePr(ctx context.Context, librariesToRelease map[string]string, pipelineState *statepb.PipelineState, repoPath string, repo *gitrepo.Repo, prDescription string) error {
+	if len(librariesToRelease) > 0 {
+		updateLibraryMetadata(librariesToRelease, pipelineState, repoPath)
+		err := saveState(repo, pipelineState)
+		if err != nil {
+			return err
+		}
+		_, err = gitrepo.AddAll(ctx, repo)
+		if err != nil {
+
+		}
+		if err := gitrepo.Commit(ctx, repo, "updating pipeline-state with latest versions"); err != nil {
+			slog.Info(fmt.Sprintf("Received error trying to commit: '%s'", err))
+			return err
+		}
+
+		err = push(ctx, repo, time.Now(), "chore(main): release", "Release "+prDescription)
+		if err != nil {
+			slog.Info(fmt.Sprintf("Received error trying to create release PR: '%s'", err))
+			return err
+		}
+	}
+	return nil
+}
+
+/*
+this goes through each library in pipeline state and checks if any new commits have been added to that library since previous commit tag
+*/
+func generateReleaseCommitForEachLibrary(ctx context.Context, repoPath string, repo *gitrepo.Repo, inputDirectory string, pipelineState *statepb.PipelineState) (map[string]string, string, error) {
+	libraries := pipelineState.LibraryReleaseStates
+	var prDescription string
+	var librariesToRelease map[string]string
+	librariesToRelease = make(map[string]string)
+	for i := 0; i < len(libraries); i++ {
+		library := libraries[i]
+		var commitMessages []string
+		//TODO: need to add common paths as well as refactor to see if can check all paths at 1 x
+		for j := 0; j < len(library.SourcePaths); j++ {
+			//TODO: figure out generic logic
+			previousReleaseTag := library.Id + "-" + library.CurrentVersion
+			commits, err := gitrepo.GetApiCommitsSinceTagForSource(repo, library.SourcePaths[j], previousReleaseTag)
+			if err != nil {
+				fmt.Println("Error searching commits:", err)
+				//TODO update PR description with this data and mark as not humanly resolvable
+			}
+			for _, commit := range commits {
+				commitMessages = append(commitMessages, commit.Message)
+			}
+		}
+
+		if len(commitMessages) > 0 && isReleaseWorthy(commitMessages) {
+			releaseNotes, err := createReleaseNotes(library, commitMessages, inputDirectory)
+			releaseVersion, err := calculateNextVersion(library)
+
+			//
+			if err := container.UpdateReleaseMetadata(flagImage, repoPath, inputDirectory, library.Id, releaseVersion); err != nil {
+				slog.Info(fmt.Sprintf("Received error running container: '%s'", err))
+
+			}
+			//TODO: make this configurable so we don't have to run per library
```

Comment:
Having fixed the .NET container so that it only builds the right library, I wouldn't expect we'd need this. Even when testing, I'd hope the build should be quick enough.